ggshield/core/git_hooks/ci/previous_commit.py

```diff
@@ -43,9 +50,18 @@ def github_previous_commit_sha(verbose: bool) -> Optional[str]:
     if pull_req_base_sha:
         return pull_req_base_sha
 
-    if push_before_sha:
+    if push_before_sha and push_before_sha != EMPTY_SHA:
         return push_before_sha
 
+    if head_sha and event_name == "push":
+        if verbose:
+            click.echo("Could not find current branch previous commit.")
+            click.echo("Current branch may have been just pushed.")
+            click.echo("Only scan last commit.")
+        last_commits = get_list_commit_SHA(f"{head_sha}~1", max_count=1)
+        if len(last_commits) == 1:
+            return last_commits[0]
```

Comment:
if this condition is not met we are raising the `UnexpectedError`, is it what we want? just to check

Reply:
Yes I took the behavior from secret, but maybe we prefer to scan ony last commit in any case where we fail to get what we want and log it ?

Reply:
However other cases should not happen: HEAD_SHA should always exist, and in case of a PR, we will have `pull_req_base_sha` defined. So as other cases seem unexpected, maybe raising in those cases is the right behavior.

Reply:
ok, fine by me!